cogs/core.py

```diff
@@ -249,20 +249,28 @@ async def acloseall(self, ctx, *, reason: str = None):
         usage="blacklist <member>",
         aliases=["block"],
     )
-    async def blacklist(self, ctx, *, member: MemberConverter):
+    async def blacklist(self, ctx, *, member: MemberConverter, reason: str = None):
         blacklist = (await tools.get_data(self.bot, ctx.guild.id))[9]
         if member.id in blacklist:
             await ctx.send(ErrorEmbed("The user is already blacklisted."))
             return
 
         async with self.bot.pool.acquire() as conn:
-            await conn.execute(
+            if len(reason) == 0:
+                await conn.execute(
                 "UPDATE data SET blacklist=array_append(blacklist, $1) WHERE guild=$2",
                 member.id,
                 ctx.guild.id,
             )
+            else:
+                await conn.execute(
+                    "UPDATE data SET blacklist=array_append(blacklist, $1, reason, $3) WHERE guild=$2",
```

Comment:
This would not work at all because the array is storing a list of blocked users. We should consider adding a separate table for blacklist.

Reply:
Fixed in #115